line.js

```diff
@@ -1,47 +1,47 @@
-/*
+/**
  * Line Utils
  * A line is an array of two points e.g. [[-5, 0], [5, 0]].
+ *
+ * @flow
  */
 
-var kpoint = require("./point.js");
-var kvector = require("./vector.js");
+import * as kpoint from "./point.js";
+import * as kvector from "./vector.js";
 
-var kline = {
+import type {Point} from "./point.js";
 
-    distanceToPoint: function(line, point) {
-        return kpoint.distanceToLine(point, line);
-    },
+export type Line = [Point, Point];
 
-    reflectPoint: function(line, point) {
-        return kpoint.reflectOverLine(point, line);
-    },
+export function distanceToPoint(line: Line, point: Point): number {
+    return kpoint.distanceToLine(point, line);
+}
 
-    midpoint: function(line) {
-        return [
-            (line[0][0] + line[1][0]) / 2,
-            (line[0][1] + line[1][1]) / 2
-        ];
-    },
+export function reflectPoint(
+    line: Line,
+    point: Point
+): $ReadOnlyArray<number> /* TODO: convert to Point */ {
```

Comment:
Is there an issue preventing this from being done right now?

Reply:
Yes. These functions use the underlying `vector` functions which are often `$ReadOnlyArray<number>` instead of a 2-number tuple.  I think if we made it explicit that this library only deals with 2D math, we could fix up all these types.